openmeter/productcatalog/plan/service/phase.go

```diff
@@ -5,140 +5,162 @@ import (
 	"fmt"
 
 	"github.com/openmeterio/openmeter/openmeter/productcatalog/plan"
+	"github.com/openmeterio/openmeter/pkg/framework/transaction"
 	"github.com/openmeterio/openmeter/pkg/pagination"
 )
 
 func (s service) ListPhases(ctx context.Context, params plan.ListPhasesInput) (pagination.PagedResponse[plan.Phase], error) {
-	if err := params.Validate(); err != nil {
-		return pagination.PagedResponse[plan.Phase]{}, fmt.Errorf("invalid list PlanPhases params: %w", err)
+	fn := func(ctx context.Context) (pagination.PagedResponse[plan.Phase], error) {
+		if err := params.Validate(); err != nil {
+			return pagination.PagedResponse[plan.Phase]{}, fmt.Errorf("invalid list PlanPhases params: %w", err)
+		}
+
+		return s.adapter.ListPhases(ctx, params)
 	}
 
-	return s.adapter.ListPhases(ctx, params)
+	return transaction.Run(ctx, s.adapter, fn)
```

Comment:
I think you need to use `entutils.TransactingRepo` no? cc @GAlexIHU

Reply:
Nope, the original idea is that each function is wrapped in adapter in the transacting repo, while in service you define the transactions. In billing I am using the transacting repo, as it achieves the same as long as the adapter is not called directly.

I might or might not change billing to have this format later.